ops/testing.py

```diff
@@ -1509,27 +1509,75 @@ def _get_backend_calls(self, reset: bool = True) -> List[Tuple[Any, ...]]:
-
-        This is named :code:`add_model_secret` instead of :code:`add_secret`
-        to avoid confusion with the :meth:`ops.Application.add_secret`
-        and :meth:`ops.Unit.add_secret` methods used by secret owner
-        charms.
+    def add_charm_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """Add a secret owned by the charm.
 
         Args:
-            owner: The name of the remote application (or specific remote
-                unit) that will own the secret.
+            owner: The name of the remote application or a specific remote unit,
+                that will own the secret.
             content: A key-value mapping containing the payload of the secret,
                 for example :code:`{"password": "foo123"}`.
 
         Return:
-            The ID of the newly-secret added.
+            The ID of the newly-added secret.
         """
         owner_name = _get_app_or_unit_name(owner)
         model.Secret._validate_content(content)
         return self._backend._secret_add(content, owner_name)
 
+    def add_model_secret(self, owner: AppUnitOrName, content: Dict[str, str]) -> str:
+        """An alias for :meth:`add_charm_secret` for backward compatibility."""
+        warnings.warn("add_model_secret is deprecated, use add_charm_secret instead.",
+                      category=DeprecationWarning)
+        return self.add_charm_secret(owner, content)
+
+    def add_user_secret(self, content: Dict[str, str]) -> str:
+        """Add a secret owned by the user, simulating the ``juju add-secret`` command.
+
+        Args:
+            content: A key-value mapping containing the payload of the secret,
+                for example :code:`{"password": "foo123"}`.
+
+        Return:
+            The ID of the newly-added secret.
+
+        Example usage (the parameter ``harness`` in the test function is
+        a pytest fixture that does setup/teardown, see :class:`Harness`)::
+
+            # config.yaml
+            options:
+              mysec:
+                type: secret
+                description: "tell me your secrets"
+
+            # charm.py
+            class MyVMCharm(ops.CharmBase):
+                def __init__(self, framework: ops.Framework):
+                    super().__init__(framework)
+                    framework.observe(self.on.config_changed, self._on_config_changed)
+
+                def _on_config_changed(self, event: ops.StartEvent):
+                    mysec = self.config.get('mysec')
+                    if mysec:
+                        sec = self.model.get_secret(id=mysec, label="mysec")
+                        self.config_from_secret = sec.get_content()
+
+            # test_charm.py
+            def test_config_changed(harness):
+                secret_content = {'password': 'foo'}
+                secret_id = harness.add_user_secret(secret_content)
+                harness.grant_secret(secret_id, 'webapp')
```

Comment:
I think we should have this as a working example (I'd like to get these doctest'd in the future). Presumably this would if you had a `charmcraft.yaml` where the name was `webapp` but by default it should be:

```suggestion
                harness.grant_secret(secret_id, 'test-charm')
```

(That's the name that Harness uses if there isn't one in the metadata).